x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -595,6 +598,10 @@ private IndexMetadata.Builder copyIndexMetadata(IndexMetadata sourceIndexMetadat
             if (FORBIDDEN_SETTINGS.contains(key)) {
                 continue;
             }
+
+            if (OVERRIDE_SETTINGS.contains(key)) {
```

Comment:
This method could be package private and even static in order to be unit tested. Perhaps in a subsequent PR?

Reply:
Maybe also need to make this static and then make `indexScopedSettings` a parameter to this method.